common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -55,7 +55,8 @@
      *
      * Threading - synchronized(this). We must support adding listeners when there is no EventExecutor.
      */
-    private Object listeners;
+    private GenericFutureListener<? extends Future<?>> listener;
+    private DefaultFutureListeners listeners;
```

Comment:
FYI we used `Object` to be able to save space in the past. That said I think this is not a good reason for this anymore

Reply:
eheh I believe so: I've reported on https://github.com/netty/netty/pull/12980#issuecomment-1308437378 the actual size by using this super nice IDEA plugin: https://plugins.jetbrains.com/plugin/10953-jol-java-object-layout